back/engines/free/email_campaigns/app/policies/email_campaigns/campaign_policy.rb

```diff
@@ -89,15 +75,7 @@ def can_access_and_modify?
     end
 
     def moderator_can_access_and_modify?
-      projects = Project.where(id: user.moderatable_project_ids)
-      if projects.any? { |p| p.visible_to == 'public' }
-        true
-      elsif record.groups.empty?
-        false
-      else
-        accessible_group_ids = GroupPolicy::Scope.new(user, Group).resolve.ids
-        (record.groups.ids - accessible_group_ids).empty?
-      end
+      user.moderatable_project_ids.include?(record.context_id) && record.manageable_by_project_moderator?
```

Comment:
To reduce the number of db queries:
```suggestion
      record.manageable_by_project_moderator? && user.moderatable_project_ids.include?(record.context_id)
```